### ISSUE TITLE
Problem when calling action from another tab

### ISSUE THREAD

**Issue by zumkorn** [logux/logux#74]
Problem when calling an action from another tab, after authentication.

1. The user opens the first tab.
2. The user click on button, call changeUser()
3. The user opens the second tab and dispatch some action.
4. We get an error in the second tab:
```ts 
Uncaught (in promise) Error: Server undid Logux action because of denied at process (index.js:230)
```
Log first tab(leader):
```
Logux state is connecting
index.js:30 Logux tab role is leader
index.js:20 Logux state is synchronized
index.js:30 Logux state is disconnected
index.js:20 Logux user ID was changed to admin
index.js:30 Logux state is connecting
index.js:20 Logux state is synchronized
index.js:20 Logux added some_action action
index.js:30 Logux state is sending
index.js:20 Logux action 1605351454740 admin:fXe6RksV:HXvolrGE 0 was undid because of denied
```
Log second tab:
```
Logux added some_action action
index.js:30 Logux state is sending
index.js:20 Logux action 1605353613354 admin:fXe6RksV:HXvolrGE 0 was undid because of denied
index.js:230  Uncaught (in promise) Error: Server undid Logux action because of denied
    at process (index.js:230)
index.js:30 Logux state is synchronized
```
Video:
[https://d.pr/v/i0oO32](https://d.pr/v/i0oO32)
Repository with example:
[https://github.com/zumkorn/logux-simple-app](https://github.com/zumkorn/logux-simple-app)

**Possible reason:**
After authentication, the old clientId remains on the first tab, which belongs to the user anonymous.
On the second tab, it is taken from localStorage.
If you change this line of code:
[https://github.com/logux/client/blob/master/client/index.js#L282](https://github.com/logux/client/blob/master/client/index.js#L282)
```ts
this.clientId = userId + ':' + this.getClientId()
```
All work well.

**Comment by ai** [comment]
What is use case to not switching user in another tab?

**Comment by zumkorn** [comment]
Usage example:

Let's assume our application is a note editor.
The user opens it in the first tab and authenticated.

Creates the first note and writes something in it. Then he wants to create another note, but wants to keep seeing this one. He opens another tab, enters the address of the application and tries to create a note.

If the user refreshes first tab, everything will work correctly. Because clientId will be taken from localStorage.

**Comment by ai** [comment]
`changeUser` API was created with idea that you will change user ID in all tabs in the same moment (for instance, by calling `changeUser` in `log.on('add', cb)` on specific action).

Otherwise, if user will reload tab with old user ID, user will see a differen result. Or if user will open any link in new tab.

**Comment by zumkorn** [comment]
The problem happens if user will open new tab after calling changeUser

**Comment by ai** [comment]
Yeap. You was right.

1. Logux Client has special localStorage key with current `clientId`. New tab is using this key to have the same clientId.
2. This localStorage key uses user ID in the name.
3. When one tab changes user ID it do not set new `clientId` key to localStorage. 
4. New tab doesn’t find old clientId in the localStorage key and generate new client ID.
5. Server denies action because server think that it was an action from another client (since client IDs are different).

```js
this.clientId = userId + ':' + this.getClientId()
```

Is the best solution since `this.getClientId()` also saving new client ID.

Can you send PR to save your name in the project’s history?

**Comment by ai** [comment]
The fix was released in 0.9.3.
